### PR TITLE
Solution to luhn excercise

### PR DESCRIPTION
**Pull request by tkerdonc**

### CODE REVIEW

File: go/luhn/luhn.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	odd digit by two, then substracts 9 to it if the result was over 9.
+	The sum of all these digits is computed, the input is considered
+	valid if said sum is divisible by 10.  */
+func Valid(input string) bool {
+	ok := false
+	digitSum := 0
+
+	spaceStrippingRegex := regexp.MustCompile(" ")
+	input = spaceStrippingRegex.ReplaceAllString(input, "")
+	isNumeric, _ := regexp.MatchString("^[0-9]", input)
```
Comment (ivajloip) [inline go/luhn/luhn.go]: are you sure that is the needed regexp? I didn't read the text of the task :blush:, but I would suppose you meant `"^[0-9]+$"`.

File: go/luhn/luhn.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	if len(input) > 1 && isNumeric {
+		for i := len(input) - 1; i >= 0; i-- {
+			currentInt := int(input[i]) - '0'
+			//Matches every second rune starting from the right
+			//might be easier to than comparing (len(input) - i)%2 to 0
+			if (len(input)-1-i)%2 == 1 {
```
Comment (ivajloip) [inline go/luhn/luhn.go]: generally modul of 2^n is quite often written (`v & ((1 << n) - 1)` :slightly_smiling_face: ). Especially for small values of `n`.

File: go/luhn/luhn.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+				}
+			}
+
+			// Summing digit in the same iteration as doublings
+			digitSum += currentInt
+		}
+
+		//performing final %10 check
+		ok = digitSum%10 == 0
+
```
Comment (ivajloip) [inline go/luhn/luhn.go]: is there a reason for this new line? I sometimes prefer to put a new line before a return statement, but I don't see the point of one before a closing bracket :slightly_smiling_face:

**Review by ivajloip — APPROVED** [review]
A couple of minor remarks, otherwise it's fine for me. Obviously it could be made more efficient if regexp is not used, but maybe this way it's more readable, so given that there are no assumptions about the expected complexity requirements, it's fine :slightly_smiling_face:

File: go/luhn/luhn.go (hunk 1 above)
Comment (tkerdonc) [reply go/luhn/luhn.go]: I absolutely did, What was I thinking ?
I am adding a new test case using the "55>"  input string. It manages to go through the false regex and yield a true result. The fix obviously does not let it through.

File: go/luhn/luhn.go (hunk 3 above)
Comment (tkerdonc) [reply go/luhn/luhn.go]: Good point, I may sometimes be overenthusiastic with newlines...

**Pull request merged**